.gitlab-ci.yml

```diff
@@ -275,16 +274,20 @@ workflow:
   rules:
     - <<: *if_main_branch
       variables:
-        USE_CACHING_PROXY_PYTHON: "false"
+        USE_CACHING_PROXY_PYTHON: "true"
+        USE_CACHING_PROXY_RUBY: "true"
     - <<: *if_release_branch
       variables:
-        USE_CACHING_PROXY_PYTHON: "false"
+        USE_CACHING_PROXY_PYTHON: "true"
+        USE_CACHING_PROXY_RUBY: "true"
     - <<: *if_deploy
       variables:
         USE_CACHING_PROXY_PYTHON: "false"
+        USE_CACHING_PROXY_RUBY: "false"
     - if: $CI_COMMIT_TAG == null
       variables:
         USE_CACHING_PROXY_PYTHON: "false"
+        USE_CACHING_PROXY_RUBY: "false"
```

Comment:
Can we not use the cache for pipelines that were triggered by other pipelines? Eg. I don't think we need the cache to be enabled for pipelines created by `integrations-core` for testing purposes.

Reply:
That's a good note ! On it

Reply:
That's a good point ! Added in [c18083d](https://github.com/DataDog/datadog-agent/pull/16259/commits/c18083d1bbf5f5b6e03dc3609a609c4a5e2f3428).